tests/unit/core/components/TestAuthenticateButton.js

```diff
@@ -132,4 +109,33 @@ describe('<AuthenticateButton />', () => {
     }));
     expect(mapStateToProps(store.getState()).isAuthenticated).toEqual(true);
   });
+
+  it('allows a signed-in user to log out', () => {
+    const handleLogOut = sinon.stub();
+
+    const root = render({ handleLogOut, isAuthenticated: true });
+    Simulate.click(root);
+
+    sinon.assert.called(handleLogOut);
+  });
+
+  describe('createHandleLogOutFunction()', () => {
+    it('gets the server to clear cookie and auth token', () => {
+      sinon.stub(api, 'logOutFromServer').returns(Promise.resolve());
+
+      const { store, state } = dispatchSignInActions();
+
+      const handleLogOut = createHandleLogOutFunction(store.dispatch);
+      const apiState = state.api;
+      // This makes sure the state contains a token because user is logged in.
+      expect(apiState.token).toBeTruthy();
+
+      return handleLogOut({ api: apiState })
+        .then(() => {
+          sinon.assert.calledWith(api.logOutFromServer, { api: apiState });
+          // This makes sure the logOutUser() action has been dispatched.
+          expect(store.getState().api.token).toBeFalsy();
```

Comment:
Why is this needed? Checking that the API was called correctly is enough, no?

Reply:
That was part of the previous test I merged and the spec reflected it so I thought that was cool to keep.

Reply:
Ah, edit: this is also because once `logOutFromServer()` has been called, we dispatch the `logOutUser()` action, and the result is that the token should be null'ed.

Reply:
Okay, I understand. It's not necessary to test that the state gets reduced because that's already covered in https://github.com/mozilla/addons-frontend/blob/071db107cb4fdac16a28071dfa60400164ed501a/tests/unit/core/reducers/test_api.js#L23

In other words, this code should rely on the interface (which is the `logOutFromServer` action), not its implementation.

You can do this by asserting that the action is dispatched:

```js
const dispatchSpy = sinon.spy(store, 'dispatch');
...
sinon.assert.calledWith(dispatchSpy, logOutFromServer());
```